nat-lab/tests/conftest.py

```diff
@@ -86,39 +90,76 @@ async def on_output(output: str) -> None:
             "tcp",
             "start=" + str(start_port),
             "num=" + str(num_ports),
-        ]).execute(on_output, on_output)
+        ]
+    elif vm_tag is ConnectionTag.MAC_VM:
+        cmd = [
+            "sysctl",
+            "-w",
+            "net.inet.ip.portrange.first=" + str(start_port),
+            "net.inet.ip.portrange.last=" + str(start_port + num_ports),
+        ]
+    else:
+        # Linux
+        cmd = [
+            "sysctl",
+            "-w",
+            "net.ipv4.ip_local_port_range="
+            + str(start_port)
+            + " "
+            + str(start_port + num_ports),
+        ]
+
+    async with new_connection_raw(vm_tag) as connection:
+        await connection.create_process(cmd).execute(on_output, on_output)
 
 
 @pytest.fixture(autouse=True)
-@pytest.mark.windows
-def setup_windows_ports(request):
+def setup_ephemeral_ports(request):
     test_name = request.node.name
 
     def execute_setup(vm_tag):
         try:
-            asyncio.run(win_ports(vm_tag))
+            asyncio.run(os_ephemeral_ports(vm_tag))
         except ProcessExecError as e:
-            print(f"win_ports_{vm_tag} process execution failed: {e}")
+            print(
+                datetime.now(),
+                f"os_ephemeral_ports_{vm_tag} process execution failed: {e}",
+            )
 
-    if "[WINDOWS_VM_" in test_name:
-        # Extract all VM numbers (WINDOWS_VM_1 and/or WINDOWS_VM_2)
+    # Handle parameterized test names
+    if (
+        "[" in test_name
+        and "]" in test_name
+        and (("DOCKER_" in test_name and "CLIENT" in test_name) or "_VM_" in test_name)
```

Comment:
also, not all test parameters will contain `DOCKER_` prefix, so will be skipped by this filter, like for example: `test_event_link_state_peers_exchanging_data_then_idling_then_resume[CONE_CLIENT_1-NEPTUN-CONE_CLIENT_2-NEPTUN1]`

Reply:
and from what I've seen, none of the tests, that run in dockers, contains `DOCKER_` prefix when printing their paramter names :smile:

Reply:
MAC_VM and WINDOWS_VM on the other hand, should be fine

Reply:
Fixed